src/amber/cli/templates/scaffold/controller/crecto/spec/controllers/{{name}}_controller_spec.cr.ecr

```diff
@@ -1,71 +1,86 @@
 <% field_name = @fields.first.name -%>
 require "./spec_helper"
 
-def create_subject
-  subject = <%= class_name %>.new
-  subject.<%= field_name %> = "test"
-  subject.save
-  subject
+def create_model
+  model = <%= class_name %>.new
+  <% if @fields.first.reference? -%>
+  model.<%= field_name %> = <%= field_name.camelcase %>.new 
+  <% else -%>
+  model.<%= field_name %> = "test"
+  <% end -%>
+  model.save
+  model
 end
 
-describe <%= class_name %>Controller do
+class <%= class_name %>ControllerTest < Spec::Controller::Test
+  getter handler : Amber::Pipe::Pipeline
+
+  def initialize
+    @handler = Amber::Server.settings.handler
+    @handler.prepare_pipelines
+  end
+end
+
+describe <%= class_name %>ControllerTest do
+  subject = <%= class_name %>ControllerTest.new
+
   Spec.before_each do
     <%= class_name %>.clear
   end
 
   describe "index" do
     it "renders all the <%= @name %>s" do
-      subject = create_subject
-      get "/<%= @name %>s"
+      model = create_model
+      response = subject.get "/<%= @name %>s"
       response.body.should contain "test"
```

Comment:
If the first field is a reference, then this will fail. Is there a different expectation that we can create? What the response body contains could be considered a view or behavior test.

The controller should return a 200 status code, a non-empty body, and I'm not sure what else.